pulpcore/cli/ansible/remote.py

```diff
@@ -0,0 +1,244 @@
+    name: str,
+    url: str,
+    **remote_options: Dict[str, Any],
+) -> None:
+    body: Dict[str, str] = {"name": name, "url": url}
+    create_remote(
+        pulp_ctx=pulp_ctx, remote_ctx=remote_ctx, body=body, remote_options=remote_options
+    )
+
+
+@remote.command(name="create")
+@click.option("--name", required=True)
+@click.option("--url", required=True)
+@click.option(
+    "--requirements-file",
+    type=click.File(),
+    help="Collections only: A Collection requirements yaml",
+)
+@click.option("--requirements", help="A string of a requirements yaml, format: '\"...\"'")
+@click.option("--auth-url", help="Collections only: URL to receive a session token")
+@click.option("--token", help="Collections only: Token key use for authentication")
+@remote_options
+@pass_entity_context
+@pass_pulp_context
+def create_collection(
+    pulp_ctx: PulpContext,
+    remote_ctx: PulpEntityContext,
+    name: str,
+    url: str,
+    requirements_file: Any,
+    requirements: Optional[str],
+    auth_url: Optional[str],
+    token: Optional[str],
+    **remote_options: Dict[str, Any],
+) -> None:
+    body: Dict[str, Any] = {"name": name, "url": url}
+
+    if requirements_file is not None:
+        body["requirements_file"] = f"{yaml.safe_load(requirements_file)}"
+    elif requirements is not None:
+        body["requirements_file"] = yaml.safe_load(f'"{requirements}"')
+    if auth_url is not None:
+        body["auth_url"] = auth_url
+    if token is not None:
+        body["token"] = token
+    create_remote(
+        pulp_ctx=pulp_ctx, remote_ctx=remote_ctx, body=body, remote_options=remote_options
+    )
+
+
+def create_remote(
+    pulp_ctx: PulpContext,
+    remote_ctx: PulpEntityContext,
+    body: Dict[str, Any],
+    remote_options: Dict[str, Any],
+) -> None:
+    if remote_options:
+        removed_nulls = {k: v for k, v in remote_options.items() if v is not None}
+        body.update(removed_nulls)
+    result = remote_ctx.create(body=body)
+    pulp_ctx.output_result(result)
+
+
+@remote.command(name="update")
+@click.option("--name", required=True)
+@click.option("--url")
+@remote_options
+@pass_entity_context
+@pass_pulp_context
+def update_role(
+    pulp_ctx: PulpContext,
+    remote_ctx: PulpEntityContext,
+    name: str,
+    url: Optional[str],
+    **remote_options: Dict[str, Any],
+) -> None:
+    """
+    Performs a partial update of the remote
+    """
+    body: Dict[str, Any] = dict()
+    if url:
+        body["url"] = url
+    update_remote(
+        pulp_ctx=pulp_ctx,
+        remote_ctx=remote_ctx,
+        name=name,
+        body=body,
+        remote_options=remote_options,
+    )
+
+
+@remote.command(name="update")
+@click.option("--name", required=True)
+@click.option("--url")
+@click.option("--requirements-file", type=click.File())
+@click.option("--requirements")
+@click.option("--auth-url")
+@click.option("--token")
+@remote_options
+@pass_entity_context
+@pass_pulp_context
+def update_collection(
```

Comment:
After thinking more about this, you should really collapse both update commands into one, and maintain a list of allowed parameters (probably in the corresponding contexts, because that is where the polymorphy lives) to error out if wrong ones are provided.